### PR TITLE
Ret 5257

### PR DESCRIPTION
**Pull request by LahiruOshara**
### Jira link

[See [PROJ-XXXXXX](https://tools.hmcts.net/jira/browse/PROJ-XXXXXX)](https://tools.hmcts.net/jira/browse/RET-5257)

### Change description

- Functionality to add multiple documents in draft and sign event

### Checklist

- [x] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: definitions/json/AuthorisationCaseField/AuthorisationCaseField-WA.json [hunk 1]
```diff
@@ -1018,5 +1018,11 @@
     "CaseFieldID": "caseManagementLocationCode",
     "UserRole": "citizen",
     "CRUD": "CRU"
+  },
+  {
+    "CaseTypeId": "ET_EnglandWales",
+    "CaseFieldID": "DraftAndSignJudgementDocuments",
+    "UserRole": "caseworker-employment-etjudge-englandwales",
+    "CRUD": "CRUD"
```
Comment (hsjhita) [inline definitions/json/AuthorisationCaseField/AuthorisationCaseField-WA.json:1026]: this is not needed

File: definitions/json/CaseEventToFields/CaseEventToFields-WA.json [hunk 2]
```diff
@@ -518,5 +518,16 @@
...
+  {
+    "CaseTypeID": "ET_EnglandWales",
+    "CaseEventID": "draftAndSignJudgement",
+    "CaseFieldID": "DraftAndSignJudgementDocuments",
+    "DisplayContext": "COMPLEX",
+    "ShowSummaryChangeOption": "Y",
+    "PageID": 1,
+    "PageDisplayOrder": 2,
+    "PageFieldDisplayOrder": 2,
+    "PageColumnNumber": 1
```
Comment (hsjhita) [inline definitions/json/CaseEventToFields/CaseEventToFields-WA.json:531]: this is not needed

File: definitions/json/CaseField/CaseField-WA.json [hunk 3]
```diff
@@ -231,17 +231,25 @@
...
     "SecurityClassification": "Public"
   },
+  {
+    "CaseTypeID": "ET_EnglandWales",
+    "ID": "DraftAndSignJudgementDocuments",
+    "Label": "Upload document",
+    "FieldType": "Collection",
+    "FieldTypeParameter": "DocumentUpload",
+    "SecurityClassification": "Public"
+  },
```
Comment (hsjhita) [inline definitions/json/CaseField/CaseField-WA.json:252]: this is not needed

File: definitions/json/ComplexTypes/ComplexTypes.json [hunk 4]
```diff
@@ -3860,16 +3860,15 @@
...
-    "ElementLabel": "Is this a Judgement",
+    "ElementLabel": "Is this a Judgment",
     "SecurityClassification": "Public"
   },
   {
     "ID": "DraftAndSignJudgement",
-    "ListElementCode": "draftAndSignJudgementDocument",
-    "FieldType": "Document",
-    "ElementLabel": "Document upload",
+    "ListElementCode": "DraftAndSignJudgementDocuments",
```
Comment (hsjhita) [inline definitions/json/ComplexTypes/ComplexTypes.json:3868]: just to make it consistent
```suggestion
    "ListElementCode": "draftAndSignJudgementDocuments",
```

File: definitions/json/EventToComplexTypes/EventToComplexTypes-WA.json [hunk 5]
```diff
@@ -704,27 +704,37 @@
...
   },
+
   {
     "ID": "DraftAndSignJudgement",
     "CaseEventID": "draftAndSignJudgement",
-    "CaseFieldID": "draftAndSignJudgement",
-    "ListElementCode": "draftAndSignJudgementDocument",
-    "EventElementLabel": "Document upload",
+    "CaseFieldID": "DraftAndSignJudgementDocuments",
+    "ListElementCode": "uploadedDocument",
```
Comment (hsjhita) [inline definitions/json/EventToComplexTypes/EventToComplexTypes-WA.json:717]: You are referencing a field which is not part of the complex type which is why it doesn't show in table. You have been saving the document to a field outside of the complex type
```suggestion
    "CaseFieldID": "draftAndSignJudgment",
    "ListElementCode": "draftAndSignJudgementDocuments",
```

File: definitions/json/EventToComplexTypes/EventToComplexTypes-WA.json [hunk 6]
```diff
@@ -704,27 +704,37 @@
...
+    "DisplayContext": "OPTIONAL"
+  },
+  {
+    "ID": "DraftAndSignJudgement",
+    "CaseEventID": "draftAndSignJudgement",
+    "CaseFieldID": "DraftAndSignJudgementDocuments",
+    "ListElementCode": "shortDescription",
+    "EventElementLabel": "Short description of document",
+    "FieldDisplayOrder": 3,
+    "DisplayContext": "OPTIONAL"
```
Comment (hsjhita) [inline definitions/json/EventToComplexTypes/EventToComplexTypes-WA.json:729]: can be removed

File: definitions/json/EventToComplexTypes/EventToComplexTypes-WA.json [hunk 7]
```diff
@@ -704,27 +704,28 @@
...
     "Publish": "Y"
   },
+
   {
     "ID": "DraftAndSignJudgement",
     "CaseEventID": "draftAndSignJudgement",
-    "CaseFieldID": "draftAndSignJudgement",
-    "ListElementCode": "draftAndSignJudgementDocument",
-    "EventElementLabel": "Document upload",
+    "CaseFieldID": "draftAndSignJudgment",
```
Comment (hsjhita) [inline definitions/json/EventToComplexTypes/EventToComplexTypes-WA.json:716]:
```suggestion
    "CaseFieldID": "draftAndSignJudgement",
```

File: definitions/json/EventToComplexTypes/EventToComplexTypes-WA.json [hunk 8]
```diff
@@ -704,27 +704,28 @@
...
     "CaseFieldID": "draftAndSignJudgement",
-    "ListElementCode": "draftAndSignJudgementDocument",
-    "EventElementLabel": "Document upload",
+    "ListElementCode": "draftAndSignJudgementDocuments",
+    "EventElementLabel": "Document",
     "FieldDisplayOrder": 2,
-    "DisplayContext": "MANDATORY"
+    "DisplayContext": "OPTIONAL"
   },
   {
```
Comment (hsjhita) [inline definitions/json/EventToComplexTypes/EventToComplexTypes-WA.json:722]: {
     "ID": "DraftAndSignJudgement",
     "CaseEventID": "draftAndSignJudgement",
     "CaseFieldID": "draftAndSignJudgement",
     "ListElementCode": "draftAndSignJudgementDocuments.uploadedDocument",
     "EventElementLabel": "Document",
     "FieldDisplayOrder": 3,
     "DisplayContext": "MANDATORY"
   },
  {
     "ID": "DraftAndSignJudgement",
     "CaseEventID": "draftAndSignJudgement",
     "CaseFieldID": "draftAndSignJudgement",
     "ListElementCode": "draftAndSignJudgementDocuments.shortDescription",
     "EventElementLabel": "Description",
     "FieldDisplayOrder": 4,
     "DisplayContext": "OPTIONAL"
   },

**Review by hsjhita — DISMISSED** [review]

File: definitions/json/CaseEvent/CaseEvent.json [hunk 9]
```diff
@@ -680,8 +680,8 @@
   {
     "CaseTypeID": "ET_EnglandWales",
     "ID": "viewAllNotifications",
-    "Name": "Judgement, Order, Notification",
-    "Description": "View all judgement, orders and notifications",
+    "Name": "Judgment, Order, Notification",
+    "Description": "View all judgment, orders and notifications",
```
Comment (hsjhita) [inline definitions/json/CaseEvent/CaseEvent.json:684]:
```suggestion
    "Description": "View all judgments, orders and notifications",
```

**Review by hsjhita — APPROVED** [review]

**Review by cindychmcts — APPROVED** [review]

**Pull request merged**